tests/rptest/tests/tombstone_removal_e2e_test.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+
+import requests
+
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.admin import Admin
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.tests.partition_movement import PartitionMovementMixin
+from ducktape.tests.test import TestContext
+
+
+class TombstoneRemovalTest(RedpandaTest):
+    topic_name = "tr_test_topic"
+    data_msg_count = 256 * 1024
+    data_msg_size = 1024
+
+    def __init__(self, test_context: TestContext):
+        extra_rp_conf = dict(
+            enable_leader_balancer=False,
+            partition_autobalancing_mode="off",
+            health_monitor_max_metadata_age=100,  # ms
+            raft_learner_recovery_rate=10 * 1024 * 1024,  # 10MB/s
+            tombstone_retention_ms=1000,  # 1 second
+        )
+        super(TombstoneRemovalTest, self).__init__(
+            num_brokers=5,
+            test_context=test_context,
+            extra_rp_conf=extra_rp_conf,
+        )
+
+    def produce_data(self, key_set_cardinality: int):
+        producer = KgoVerifierProducer(
+            self.test_context,
+            self.redpanda,
+            self.topic_name,
+            msg_size=self.data_msg_size,
+            msg_count=self.data_msg_count,
+            key_set_cardinality=key_set_cardinality,
+            tombstone_probability=0.0,
+        )
+        try:
```

Comment:
What's the intention here? Do you want to send all messages or 100?

If the former why you set high msg_count and not just 100? You can also just `KgoVerifierProducer.oneshot([params from above as-is])`

If the later, you need to call `wait()` not `stop()` I believe.

Reply:
I want to send all messages, and wait for acks for no longer than 100 seconds. I'll see if I can simplify it using oneshot, thanks

Reply:
Ah. Weird to see wait_for_acks when you want to wait for all messages (i assumed 100 means number of acks by mistake). oneshot or wait() is more readable.

wait_for_acks is often used to say send unbounded messages but you want only for a small number to be delivered before continuing with the test to make sure that data is flowing.